language/stdlib/modules/libra_system_2.move

```diff
@@ -0,0 +1,264 @@
+address 0x0:
+
+// These modules only change resources stored under the three addresses:
+// LibraAssociation's address: 0xA550C18
+// ValidatorSet address: 0x1D8
+// DiscoverySet address: 0xD15C0
+// ValidatorSet and DiscoverySet are stored under different addresses in order to facilitate cheaper reads of these sets from storage. The keys to these addresses are also owned by the Libra Association entity, therefore the caller on all of the methods is the LibraAssociation's address.
+module LibraSystem2 {
+    use 0x0::LibraAccount;
+    use 0x0::ValidatorConfig2;
+    use 0x0::Vector;
+    use 0x0::Transaction;
+    use 0x0::LibraTimestamp;
+
+    struct ValidatorInfo {
+        addr: address,
+        consensus_voting_power: u64,
+        config: ValidatorConfig2::Config,
+        // To protect against DDoS, the force updates are allowed
+        // only if the time passed since the last force update is larger than a threshold (24hr).
+        last_force_update_time: u64,
+    }
+    
+    struct ValidatorSetChangeEvent {
+        new_validator_set: vector<ValidatorInfo>,
+    }
+
+    resource struct ValidatorSet {
+        validators: vector<ValidatorInfo>,
+        last_reconfiguration_time: u64,
+        change_events: LibraAccount::EventHandle<ValidatorSetChangeEvent>,
+    }
+    
+    // ValidatorInfo public accessors
+    public fun get_validator_address(v: &ValidatorInfo): &address { &v.addr }
+    public fun get_validator_config(v: &ValidatorInfo): &ValidatorConfig2::Config { &v.config }
+    public fun get_validator_voting_power(v: &ValidatorInfo): &u64 { &v.consensus_voting_power }
+    public fun get_last_force_update_time(v: &ValidatorInfo): &u64 { &v.last_force_update_time }
+
+    // This can only be invoked by the ValidatorSet address to instantiate the resource under that address.
+    // It can only be called a single time. Currently, it is invoked in the genesis transaction.
```

Comment:
can you imagine how one might initialize this otherwise? like we don't have the account key for this address... I think we should try to imagine a way in which all accounts that we store data can somehow be manipulated post genesis... but may be I am a dreamer.

cc: @sblackshear

Reply:
It would be fine to initialize this after genesis if it were anything other than the validator set... But we can certainly set it up so it can be manipulated after genesis. What do you have in mind?